### PR TITLE
Fix(Data Mapper): fixing edifact nested loop mapping bug 

### PR DESCRIPTION
**Pull request by DanielleCogs**
- **Please check if the PR fulfills these requirements**

* [x] The commit message follows our guidelines
* [x] Tests for the changes have been added (for bug fixes/features)
* [x] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix
- **What is the current behavior?** (You can also link to an open issue here)
- see attached bug
- **What is the new behavior (if this is a feature change)?**
allow backout of current loop using '../'
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no
- **Please Include Screenshots or Videos of the intended change**:

### CODE REVIEW

File: libs/data-mapper/src/lib/utils/DataMap.Utils.ts [hunk 1]
```diff
@@ -406,6 +426,7 @@ export const Separators = {
   CloseParenthesis: ')',
   Comma: ',',
   Dollar: '$',
+  //ForwardSlash: '/'
```
Comment (Elaina-Lee) [inline libs/data-mapper/src/lib/utils/DataMap.Utils.ts:429]: Would this be for future reference?

File: libs/data-mapper/src/lib/mapDefinitions/MapDefinitionSerializer.ts [hunk 2]
```diff
@@ -109,103 +113,151 @@ export const generateMapDefinitionBody = (mapDefinition: MapDefinitionEntry, con
...
+            if (
+              formattedLmlSnippetForSource === inputNode.node.key &&
+              inputNode.node.nodeProperties.includes(SchemaNodeProperty.Repeating)
+            ) {
+              formattedLmlSnippetForSource = '.';
             } else if (valueToTrim) {
-              value = value.replace(`${valueToTrim}/`, '');
+              // account for source elements at different level of loop
+              let backoutValue = '';
+              if (valueToTrim !== lastLoop.loop && !valueToTrim.includes('/*')) {
```
Comment (Elaina-Lee) [inline libs/data-mapper/src/lib/mapDefinitions/MapDefinitionSerializer.ts:234]: In which case would a part of data map value include '/*'?

**Review by Elaina-Lee — APPROVED** [review]

File: libs/data-mapper/src/lib/utils/DataMap.Utils.ts (hunk 1 above)
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/utils/DataMap.Utils.ts:429]: yeah I think we will need it soon!

File: libs/data-mapper/src/lib/mapDefinitions/MapDefinitionSerializer.ts (hunk 2 above)
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/mapDefinitions/MapDefinitionSerializer.ts:234]: this is for json arrays! And there is a bug in the JSON array serialization, so this feature won't work for json for now

**Pull request merged**